### ISSUE TITLE
cpp-driver should default to Advanced Shard Awareness

### ISSUE THREAD

**Issue by fee-mendes** [scylladb/cpp-driver#65]
**TLDR;** it makes no sense to have a recommended feature that is not the default

It's been long since Scylla <4.3 went out of support, back from a time when it made sense to have [basic shard-awareness](https://github.com/scylladb/cpp-driver/tree/master/topics/scylla_specific#basic-shard-awareness) by default. The world has evolved since then, and [Advanced Shard Awareness](https://github.com/scylladb/cpp-driver/tree/master/topics/scylla_specific#advanced-shard-awareness) was introduced, which is able to fallback to basic shard-awareness mode whenever needed.

One potential argument against not enabling it by default would be the need to set `cass_cluster_set_local_port_range(start, end)` ephemeral port range. I advocate that this parameter should either default to the value of `net.ipv4.ip_local_port_range` or - if we want something portable - `49152-65535` seems to be the standard "informal agreement" across most platforms.

@jul-stas WDYT?
